Sming/Components/libsodium/component.mk

```diff
@@ -0,0 +1,103 @@
+# umm_malloc (custom heap allocation)
+
+COMPONENT_SUBMODULES	:= libsodium
+
+LIBSODIUM_ROOT := $(COMPONENT_PATH)/libsodium
+LIBSODIUM_SRC := $(LIBSODIUM_ROOT)/src/libsodium
+
+COMPONENT_SRCFILES := \
```

Comment:
If you wanted to avoid listing all files individually, see `Sming/component.mk` for use of ListAllSubDirs.

Reply:
In this case the file listing is intentional, because I excluded all files with architecture-specific assembly-optimized implementations.